drivers/storage/portworx/portworx.go

```diff
@@ -629,12 +629,15 @@ func setPortworxDefaults(toUpdate *corev1.StorageCluster) {
 		}
 	}
 
-	// Enable CSI flag by default
-	if _, ok := toUpdate.Spec.FeatureGates[string(pxutil.FeatureCSI)]; !ok {
-		if toUpdate.Spec.FeatureGates == nil {
-			toUpdate.Spec.FeatureGates = make(map[string]string)
+	if t.isK3s || toUpdate.Spec.Version == "" {
```

Comment:
So k3s cannot use in-tree and only uses CSI?

Reply:
Correct. In k3s in-tree plugins are disabled to make k8s lighter. CSI is the only way to interact with storage providers.